suite/tests/api/ir_x86.c

```diff
@@ -787,6 +787,46 @@ test_instr_encode_and_decode(void *dc, instr_t *instr, uint len_expect,
     instr_destroy(dc, decin);
 }
 
+bool
+instr_same_synthetic(instr_t *inst1, instr_t *inst2)
+{
+    if (instr_num_srcs(inst1) != instr_num_srcs(inst2))
+        return false;
+
+    if (instr_num_dsts(inst1) != instr_num_dsts(inst2))
+        return false;
+
+    if (instr_get_isa_mode(inst1) != instr_get_isa_mode(inst2))
+        return false;
+
+    return true;
+}
+
+static void
+test_instr_encode_decode_synthetic(void *dc, instr_t *instr, byte *bytes)
+{
+    instr_t *instr_copy;
+    instr_set_isa_mode(instr, DR_ISA_SYNTH);
+    instr_encode(dc, instr, bytes);
+    instr_copy = instr_create(dc);
+    instr_set_isa_mode(instr_copy, DR_ISA_SYNTH);
+    decode(dc, bytes, instr_copy);
+    ASSERT(instr_same_synthetic(instr, instr_copy));
+    instr_destroy(dc, instr);
+    instr_destroy(dc, instr_copy);
+}
+
+static void
+test_instr_create_encode_decode_synthetic(void *dc)
+{
+    instr_t *instr;
+    opnd_t abs_addr = opnd_create_abs_addr((void *)0xdeadbeefdeadbeef, OPSZ_8);
+
+    instr = INSTR_CREATE_mov_ld(dc, opnd_create_reg(DR_REG_RAX), abs_addr);
+    byte bytes[8];
```

Comment:
How do you know it fits in 8 bytes?

Can have the test check the size.

Reply:
As in: checking instr_t length?

Reply:
Calling `instr_length()` (should prob also ensure it equals the PC differences for the encode and decode routines).

Reply:
Done.
Now checking that next_pc for encode and decode are the same, since the instr length should be the same.